crates/biome_service/src/settings.rs

```diff
@@ -782,35 +796,75 @@ impl VcsSettings {
 #[derive(Clone, Debug)]
 pub enum VcsIgnoredPatterns {
     Git {
-        // Represents the `.gitignore` file at the root of the project
+        /// Represents the `.gitignore` file at the root of the project
         root: Gitignore,
-        // The list of nested `.gitignore` files found inside the project
+        /// The list of nested `.gitignore` files found inside the project
         nested: Vec<Gitignore>,
     },
 }
 
 impl VcsIgnoredPatterns {
-    pub fn is_ignored(&self, path: &Utf8Path, is_dir: bool) -> bool {
+    /// Checks whether the path ignored by any ignore file found inside the project
+    ///
+    /// The `root_path` represents the root of the project, as we want to match all ignore files untile the root.
+    pub fn is_ignored(&self, root_path: &Utf8Path, path: &Utf8Path, is_dir: bool) -> bool {
         match self {
-            Self::Git { root, nested } => {
-                root.matched(path, is_dir).is_ignore()
-                    || nested.iter().any(|gitignore| {
-                        let ignore_directory = if gitignore.path().is_file() {
-                            // SAFETY: if it's a file, it always has a parent
-                            gitignore.path().parent().unwrap()
-                        } else {
-                            gitignore.path()
-                        };
-                        if let Ok(stripped_path) = path.strip_prefix(ignore_directory) {
-                            gitignore.matched(stripped_path, is_dir).is_ignore()
+            Self::Git { root, nested, .. } => {
+                // NOTE: this could be a bug of the library, need to explore. Let's assume it isn't
+                // When crawling the file system with the CLI, we correctly exclude ignored folders
+                // such as `dist/` or `build/`, in case the path to match is `/Users/foo/project/dist`
+                //
+                // However, the LSP sends absolute file paths, e.g. `/Users/foo/project/dist/a.min.js`,
+                // and they **don't** match globs such as `dist/`.
+                // To work around this limitation, we crawl upwards the parents of the path, until
+                // we arrive at the `root_path`.
+                if path.is_file() && path.is_absolute() {
+                    let mut current_path = path;
+                    let mut is_ignore = false;
+                    loop {
+                        if current_path == root_path {
+                            break;
+                        }
+                        if Self::is_git_ignore(root, nested.as_slice(), current_path, is_dir) {
+                            is_ignore = true;
+                            break;
+                        }
```

Comment:
You can avoid the mutable variable here by forcing a return type on the `loop`. The advantage here is that the compiler will also enforce that any `break` has a correct value, and you can't accidentally continue the loop after setting `is_ignore` to `true`:

```suggestion
                    let is_ignore = loop {
                        if current_path == root_path {
                            break false;
                        }
                        if Self::is_git_ignore(root, nested.as_slice(), current_path, is_dir) {
                            break true;
                        }
```